src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -803,14 +863,388 @@ def forward(
+    """
+
+    model_type: str = "encodec_discriminator"
+    filters: int = 32
+    in_channels: int = 1
+    out_channels: int = 1
+    n_ffts: list = field(default_factory=lambda: [1024, 2048, 512])
+    hop_lengths: list = field(default_factory=lambda: [256, 512, 128])
+    win_lengths: list = field(default_factory=lambda: [1024, 2048, 512])
+    kernel_size: tuple = (3, 9)
+    stride: tuple = (1, 2)
+    dilations: list = field(default_factory=lambda: [1, 2, 4])
+    max_filters: int = 1024
+    filters_scale: int = 2
+    normalized: bool = True
+    norm: str = "weight_norm"
+    activation: str = "LeakyReLU"
+    activation_params: dict = field(default_factory=lambda: {"negative_slope": 0.2})
+    output_hidden_states: bool = False
+    output_attentions: bool = False
+    return_dict: bool = False
+    torchscript: bool = False
+
+
+class ConvLayerNorm(nn.LayerNorm):
+    """
+    Convolution-friendly LayerNorm that moves channels to last dimensions
+    before running the normalization and moves them back to original position right after.
+    """
+
+    def __init__(self, normalized_shape: tp.Union[int, tp.List[int], torch.Size], **kwargs):
+        super().__init__(normalized_shape, **kwargs)
+
+    def forward(self, x):
+        # Move time dimension to second position
+        ndim = len(x.shape)
+        perm = [0] + [ndim - 1] + list(range(1, ndim - 1))
+        x = x.permute(perm)
+
+        x = super().forward(x)
+
+        # Move time dimension back to last position
+        perm = [0] + list(range(2, ndim)) + [1]
+        x = x.permute(perm)
+        return x
+
+
+CONV_NORMALIZATIONS = frozenset(
+    ["none", "weight_norm", "spectral_norm", "time_layer_norm", "layer_norm", "time_group_norm"]
+)
+
+
+def apply_parametrization_norm(module: nn.Module, norm: str = "none") -> nn.Module:
+    assert norm in CONV_NORMALIZATIONS
+    if norm == "weight_norm":
+        return weight_norm(module)
+    elif norm == "spectral_norm":
+        return spectral_norm(module)
+    else:
+        # We already check was in CONV_NORMALIZATION, so any other choice
+        # doesn't need reparametrization.
+        return module
+
+
+def get_norm_module(module: nn.Module, causal: bool = False, norm: str = "none", **norm_kwargs) -> nn.Module:
+    """Return the proper normalization module. If causal is True, this will ensure the returned
+    module is causal, or return an error if the normalization doesn't support causal evaluation.
+    """
+    assert norm in CONV_NORMALIZATIONS
+    if norm == "layer_norm":
+        assert isinstance(module, nn.modules.conv._ConvNd)
+        return ConvLayerNorm(module.out_channels, **norm_kwargs)
+    elif norm == "time_group_norm":
+        if causal:
+            raise ValueError("GroupNorm doesn't support causal evaluation.")
+        assert isinstance(module, nn.modules.conv._ConvNd)
+        return nn.GroupNorm(1, module.out_channels, **norm_kwargs)
+    else:
+        return nn.Identity()
```

Comment:
Is `norm` always the same for the Discriminator ? it it is, let's remove this code and always use the default value!